File: Kotlin/프로그래머스/0/120838. 모스부호 （1）/모스부호 （1）.kt
class Solution {
    private val morse: Map<String, String> = mapOf(".-" to "a", "-..." to "b", "-.-." to "c", "-.." to "d", "." to "e", "..-." to "f", "--." to "g", "...." to "h", ".." to "i", ".---" to "j", "-.-" to "k", ".-.." to "l", "--" to "m", "-." to "n", "---" to "o", ".--." to "p", "--.-" to "q", ".-." to "r", "..." to "s", "-" to "t", "..-" to "u", "...-" to "v", ".--" to "w", "-..-" to "x", "-.--" to "y", "--.." to "z")
    fun solution(letter: String): String {
        return letter.split(' ')
            .map { word ->
                morse[word]
            }.joinToString("")
    }
}

File: Go/프로그래머스/0/120838. 모스부호 （1）/모스부호 （1）.go
import "strings"

var morse = map[string]string {
	".-": "a", "-...": "b", "-.-.": "c", "-..": "d", ".": "e", "..-.": "f", 
	"--.": "g", "....": "h", "..": "i", ".---": "j", "-.-": "k", ".-..": "l", 
	"--": "m", "-.": "n", "---": "o", ".--.": "p", "--.-": "q", ".-.": "r", 
	"...": "s", "-": "t", "..-": "u", "...-": "v", ".--": "w", "-..-": "x", 
	"-.--": "y", "--..": "z",
}

func solution(letter string) string {
    words := strings.Split(letter, " ")
    
    var answer strings.Builder
    
    for _, word := range words {
        converted := morse[word]
        answer.WriteString(converted)
    }
    
    return answer.String()
}